pkg/plugins/policies/meshpassthrough/plugin/xds/order.go

```diff
@@ -11,108 +11,136 @@ import (
 
 	core_mesh "github.com/kumahq/kuma/pkg/core/resources/apis/mesh"
 	api "github.com/kumahq/kuma/pkg/plugins/policies/meshpassthrough/api/v1alpha1"
+	"github.com/kumahq/kuma/pkg/util/pointer"
 )
 
 type MatchType int
 
 const (
-	Domain MatchType = iota + 1
-	WildcardDomain
-	IP
-	IPV6
+	WildcardDomain MatchType = iota + 1
+	Domain
 	CIDR
 	CIDRV6
+	IP
+	IPV6
 )
 
+var protocolOrder = map[core_mesh.Protocol]int{
+	core_mesh.ProtocolTLS:   0,
+	core_mesh.ProtocolTCP:   1,
+	core_mesh.ProtocolHTTP:  2,
+	core_mesh.ProtocolHTTP2: 2,
+	core_mesh.ProtocolGRPC:  2,
+}
+
 type Route struct {
 	Value     string
 	MatchType MatchType
 }
 
 type Matcher struct {
-	Protocol core_mesh.Protocol
-	Port     uint32
+	Protocol  core_mesh.Protocol
+	Port      uint32
+	MatchType MatchType
+	Value     string
 }
 
-type FilterChainMatcher struct {
-	Protocol core_mesh.Protocol
-	Port     uint32
-	Routes   []Route
+type FilterChainMatch struct {
+	Protocol  core_mesh.Protocol
+	Port      uint32
+	MatchType MatchType
+	Value     string
+	Routes    []Route
 }
```

Comment:
One idea from my mind but not sure if it works well. Since this PR would supplement `MatchType` and `Value` properties for `Matcher` type, which leads to a little bit coupled for type `Route` and `Matcher`.

Not sure if it would be more simple for the implementation with the following structure:

```text
type NewFilterChainMatcher struct {
	Protocol core_mesh.Protocol
	Port     uint32
	Routes   []string
}

var dic = map[MatchType][]NewFilterChainMatcher{}
```

Then just return the `dic` for function `GetOrderedMatchers`

For `listeners_filter_chain.go` structure `FilterChainConfigurer`, we may just use

```text
type FilterChainConfigurer struct {
	APIVersion core_xds.APIVersion
	MatchType MatchType
        FilterChainMatchers []NewFilterChainMatcher{}
	IsIPv6     bool
}
```

Reply:
I think we should create an issue to refactor it and make it more readable.